### PR TITLE
Authenticator

### PR DESCRIPTION
**Pull request by dpanayotov**
This provides just the mechanism to decide whether authentication is valid or not. Filter implementation is not in the scope of this issue

### CODE REVIEW

File: authentication/oidc/odic.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+	var p providerJSON
+	err = json.Unmarshal(body, &p)
+	if err != nil {
+		return nil, err
+	}
+
+	keySet := oidc.NewRemoteKeySet(ctx, p.JWKSURL)
+	cfg := &oidc.Config{
+		ClientID: options.ClientID,
```
Comment (pankrator) [inline authentication/oidc/odic.go]: Don't we need client_secret also?

File: authentication/oidc/oidc.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	req, err := http.NewRequest(http.MethodGet, wellKnown, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var readConfigFunc DoRequestFunc
+	if options.ReadConfigurationFunc != nil {
+		readConfigFunc = options.ReadConfigurationFunc
+	} else {
+		readConfigFunc = http.DefaultClient.Do
```
Comment (pankrator) [inline authentication/oidc/oidc.go]: Default client does not have request timeouts set. We should go against the option to use it by default

Comment (dpanayotov) [reply authentication/oidc/oidc.go]: I think it's nice to have defaults. However, I strongly agree the default shouldn't be used in production.

Comment (pankrator) [reply authentication/oidc/oidc.go]: Actually, if a request with context that has timeouts set is provided it will be sufficient

**Pull request merged**